.github/workflows/build-multiarch.yml

```diff
@@ -50,5 +70,20 @@ jobs:
           buildah manifest add ${{ inputs.name }} ${{ secrets.registry }}/${{ inputs.name }}:${{ inputs.tag }}-linux-s390x
 
       - name: Push manifest
+        id: push-manifest
+        run: |
+            podman manifest push --digestfile imagedigest ${{ inputs.name }} ${{ secrets.registry }}/${{ inputs.name }}:${{ inputs.tag }}  --all
+            echo "digest=$(cat imagedigest)" | tee -a $GITHUB_OUTPUT
+
+      - name: Sign the published manifest
+        if: ${{ inputs.sign == true }}
+        run: |
+          cosign sign --yes --recursive ${{ secrets.registry }}/${{ inputs.name }}@${{ steps.push-manifest.outputs.digest }}
+
+      - name: Verify the image signature
+        if: ${{ inputs.sign == true }}
         run: |
-            podman manifest push ${{ inputs.name }} ${{ secrets.registry }}/${{ inputs.name }}:${{ inputs.tag }}  --all
+          cosign verify \
+            --certificate-identity https://github.com/redhat-openshift-ecosystem/openshift-preflight/.github/workflows/build-multiarch.yml@refs/tags/${{ inputs.tag }} \
+            --certificate-oidc-issuer https://token.actions.githubusercontent.com \
+            quay.io/opdev/preflight:${{ inputs.tag }}
```

Comment:
Nit: no new line here

Reply:
I have this newline ready to push, but GitHub is having issues with actions. I'll push that when it's all resolved because it'll retrigger all testing.